rehydrate/fargate/dataset.go

```diff
@@ -59,16 +60,21 @@ func (dr *DatasetRehydrator) rehydrate(ctx context.Context) (*RehydrationResult,
 		if err != nil {
 			return nil, err
 		}
+		urlEscapedPath := utils.CreateURLEscapedPath(j.Path)
 		datasetFileByVersionResponse, err := dr.pennsieveClient.Discover.GetDatasetFileByVersion(
-			ctx, dataset32, version32, j.Path)
+			ctx, dataset32, version32, urlEscapedPath)
 		if err != nil {
 			return nil, fmt.Errorf("error retrieving dataset file %s by version: %w", j.Path, err)
 		}
+		awsEscapedPath := utils.CreateAWSEscapedPath(j.Path)
```

Comment:
I don't think this is where the AWS escaped path is needed. Neither for NewSourceObject or for DestinationKey.

SourceObject has a field called CopySource and I think that is where the escaping is required. This is the field that gets passed to the AWS input structs as CopySource, not the other path fields.

The implementation of NewSourceObject uses another util method, `VersionedCopySource()` to create this CopySource. So I think that is where the AWS escaping should be applied. Specifically on the `u.Path` component in that method. (`u.Host` is the bucket name for these uris and so should not need escaping. Same for the S3 version id part.)

Reply:
updated @jfromme